frontend/src/pages/modelRegistrySettings/ModelRegistryTableRowStatus.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+
+import { Label, Popover, Stack, StackItem } from '@patternfly/react-core';
+import {
+  CheckCircleIcon,
+  DegradedIcon,
+  ExclamationCircleIcon,
+  ExclamationTriangleIcon,
+  InProgressIcon,
+} from '@patternfly/react-icons';
+
+import { K8sCondition } from '~/k8sTypes';
+
+enum ModelRegistryStatus {
+  Progressing = 'Progressing',
+  Degraded = 'Degraded',
+  Available = 'Available',
+  IstioAvailable = 'IstioAvailable',
+  GatewayAvailable = 'GatewayAvailable',
+}
+
+enum ModelRegistryStatusLabel {
+  Progressing = 'Progressing',
+  Available = 'Available',
+  Degrading = 'Degrading',
+  Unavailable = 'Unavailable',
+}
+
+enum ConditionStatus {
+  True = 'True',
+  False = 'False',
+}
+
+interface ModelRegistryTableRowStatusProps {
+  conditions: K8sCondition[] | undefined;
+}
+
+export const ModelRegistryTableRowStatus: React.FC<ModelRegistryTableRowStatusProps> = ({
+  conditions,
+}) => {
+  const conditionsMap =
+    conditions?.reduce((acc: Record<string, K8sCondition | undefined>, condition) => {
+      acc[condition.type] = condition;
+      return acc;
+    }, {}) ?? {};
+  let statusLabel: string = ModelRegistryStatusLabel.Progressing;
+  let icon = <InProgressIcon />;
+  let color: React.ComponentProps<typeof Label>['color'] = 'blue';
+  let popoverMessages: string[] = [];
+  let popoverTitle = '';
+
+  if (Object.values(conditionsMap).length) {
+    const {
+      [ModelRegistryStatus.Available]: availableCondition,
+      [ModelRegistryStatus.Progressing]: progressCondition,
+      [ModelRegistryStatus.Degraded]: degradedCondition,
+    } = conditionsMap;
+    const lastAvailableConditionTime = new Date(
+      availableCondition?.lastTransitionTime ?? '',
+    ).getTime();
+
+    popoverMessages =
+      availableCondition?.status === ConditionStatus.False
+        ? Object.values(conditionsMap).reduce((messages: string[], condition) => {
```

Comment:
We want to show the messages from any conditions with `ConditionStatus.False` even if they aren't in the conditionsMap, since there may be additional condition types added in the future that affect availability. I think instead of reducing `Object.values(conditionsMap)` you can just reduce the whole `conditions` array here.

Reply:
`conditionsMap` uses all of the `conditions`, so I'm confused as to why this is a desired change.

Reply:
Nevermind, I misread the code. Thanks.